src/mv_memory.rs

```diff
@@ -35,8 +33,8 @@ pub struct MvMemory {
     pub(crate) data: DashMap<MemoryLocationHash, BTreeMap<TxIdx, MemoryEntry>, BuildIdentityHasher>,
     /// Last read & written locations of each transaction
     last_locations: Vec<Mutex<LastLocations>>,
-    /// Lazy addresses that need full evaluation at the end of the block
-    lazy_addresses: Mutex<LazyAddresses>,
+    /// Lazy locations that need full evaluation at the end of the block
+    lazy_locations: Mutex<HashSet<MemoryLocation, BuildSuffixHasher>>,
```

Comment:
1. Now that we have more lazy locations `DashSet` may be faster than `Mutex<HashSet<..>>` now. Worth trying a bench.
2. Using `BuildSuffixHasher` for `MemoryLocation` is pretty risky since not every possible value is an existing hash (of the same input distribution). I'd just use `ahash` for now.

Reply:
I am trying `DashSet`.

Reply:
In the latest benchmark, `DashSet` causes the performance to drop from 1.91 to 1.90.

So, it's inconclusive to say which one is better. Personally, using`DashSet` feels more comfortable.